build-logic/binary-compatibility/build.gradle.kts

```diff
@@ -25,3 +27,9 @@ dependencies {
 tasks.compileGroovy.configure {
     classpath += files(tasks.compileKotlin)
 }
+
+tasks.withType<KotlinCompile>().configureEach {
```

Comment:
Would be good to have some kind of TODO or reminder here too, to revert this at some point.

Reply:
I point here from https://github.com/gradle/gradle-private/issues/4287 already, this should be enough